federation/cmd/federation-controller-manager/app/options/options.go

```diff
@@ -115,5 +118,6 @@ func (s *CMServer) AddFlags(fs *pflag.FlagSet) {
 	fs.IntVar(&s.APIServerBurst, "federated-api-burst", s.APIServerBurst, "Burst to use while talking with federation apiserver")
 	fs.StringVar(&s.DnsProvider, "dns-provider", s.DnsProvider, "DNS provider. Valid values are: "+fmt.Sprintf("%q", dnsprovider.RegisteredDnsProviders()))
 	fs.StringVar(&s.DnsConfigFile, "dns-provider-config", s.DnsConfigFile, "Path to config file for configuring DNS provider.")
+	fs.BoolVar(&s.EnableIngressController, "enable-ingress-controller", s.EnableIngressController, "Enable federation ingress controller.")
```

Comment:
This way we will need to add a flag for each resource.
How about making this a comma separated list of resources whose controllers should be disabled.
It will be empty by default. Users who do not want ingress controller can add "ingress" to that list.